NachoSpigot-Server/src/main/java/net/minecraft/server/EntityLiving.java

```diff
@@ -1789,7 +1790,28 @@ public void receive(Entity entity, int i) {
     }
 
     public boolean hasLineOfSight(Entity entity) {
-        return this.world.rayTrace(new Vec3D(this.locX, this.locY + (double) this.getHeadHeight(), this.locZ), new Vec3D(entity.locX, entity.locY + (double) entity.getHeadHeight(), entity.locZ)) == null;
+        if (entity instanceof EntityPlayer && NachoConfig.enableImprovedHitReg) {
+            // Head height is 1,5725
+            // Split it into three to get a more accurate line of sight -> 0.52416667
+
+            double parts = entity.getHeadHeight() / 3;
+
+            return this.world.rayTrace(
```

Comment:
This looks fine to me, but was this "benchmarked" in some way? Is it rare for the 3 ray traces to get called, or do they all get called every time? Since then this might get expensive. I'm not sure, you can ignore this comment for now.

Reply:
No, I haven't benchmarked it. From a really general viewpoint, the new changes do not affect the server performance at all. I've been running an older patch for this hit detection system with some optimization issues and it runs absolutely fine. Didn't cause any lag but I personally haven't tested it with higher player counts.